edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ServiceProxy.cs

```diff
@@ -166,6 +169,21 @@ public async Task<Option<ServiceIdentity>> GetServiceIdentity(string deviceId, s
             return serviceIdentityResult;
         }
 
+        Exception MapException(DeviceScopeApiException ex)
```

Comment:
As discussed, not sure if upstream will return the same response as Hub's. At least manual test is required for nested edge scenario.

Reply:
tested in nested env and it works with amqp or mqtt older protocol head, with new mqtt broker in nested env it won't detect when there is unauthorized so tracking that as a separate issue that can be solved with MQTT5 support.